R/calculate_coaches_vote_possibilities.R

```diff
@@ -0,0 +1,175 @@
+#' Calculate Coaches Vote Possibilities
+#'
+#' @description
+#' `calculate_coaches_vote_possibilities` returns all possible breakdowns of coaches votes between two coaches,
+#' given a breakdown of coaches votes
+#'
+#' @param df Requires the following column names: Player.Name, Coaches.Votes. These can be returned from the
+#' function `fetch_coaches_votes`.
+#' @param output_type One of "Coach View", "Player View". Defaults to "Coach View".
+#'
+#' @return
+#' For output_type "Coach View" - A list of data frames with columns: Votes, C1, C2
+#' For output_type "Player View" - A data frame with columns: Player, V1, V2
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' # Return coaches votes for a particular match, then find the possibilities
+#' df <- fetch_coaches_votes(comp = "AFLM", season = 2021, round = 24, team = "Western Bulldogs")
+#' calculate_coaches_vote_possibilities(df, "Coach View")
+#' 
+#' df <- fetch_coaches_votes(comp = "AFLW", season = 2021, round = 9, team = "Western Bulldogs")
+#' calculate_coaches_vote_possibilities(df, "Player View")
+#' 
+#' # Create a manual data frame to calculate possibilities
+#' df <- data.frame(
+#' Player.Name = c("Tom Liberatore","Jack Macrae","Marcus Bontempelli","Cody Weightman","Darcy Parish","Aaron Naughton","Jordan Ridley"),
+#' Coaches.Votes = c(7, 6, 5, 5, 4, 2, 1)
+#' )
+#' calculate_coaches_vote_possibilities(df, "Player View")
+#' }
+#'
+#' @importFrom magrittr %>%
+#' @importFrom rlang .data
+calculate_coaches_vote_possibilities <- function(df, output_type){
+  
+  require(tidyverse)
```

Comment:
We need to remove this `require` call. The way to ensure packages are included is by adding them to the DESCRIPTION file. We generally avoid having tidyverse in that file and instead include the actual packages you need (eg dplyr, tidyr etc). Most of them should already be there.

Once a package is listed in the DESCRIPTION file, you can use it in the code. To do that - for each function you generally have to use the package name in the code.

e.g. `dplyr::mutate()` 

You should be able to see that in some of the other files in repo but let me know if that doesn't make sense!